lib/ansible/modules/windows/win_psmodule.ps1

```diff
@@ -79,15 +79,17 @@ Function Install-PrereqModule {
                         $install_params.SkipPublisherCheck = $true
                     }
 
-                    Install-Module @install_params > $null
+                    $moduleInstalled = Get-InstalledModule -Name $install_params.Name -MinimumVersion $install_params.MinimumVersion -ErrorAction Ignore
 
-                    if ( $Name -eq 'PowerShellGet' ) {
-                        # An order has to be reverted due to dependency
-                        Remove-Module -Name PowerShellGet, PackageManagement -Force
-                        Import-Module -Name PowerShellGet, PackageManagement -Force
+                    if($null -ne $moduleInstalled) {
+                        Install-Module @install_params > $null
+                    }
+                    else {
+                        Update-Module @install_params > null
```

Comment:
According to https://docs.microsoft.com/en-us/powershell/module/powershellget/update-module?view=powershell-6, `Update-Module` does not support the `MinimumVersion` parameter so this would fail. It does have a `-RequiredVersion` which sounds like something we may want to use instead here.

Reply:
Yes, but you also need to consider to be able to work with PowerShell 5.1 as it is mostly used on the machines still

Reply:
The -AllowClobber overwrite just methods of different modules, but Update updates the module complete which is better in terms of consistency

Reply:
"Overrides warning messages about installation conflicts about existing commands on a computer. Overwrites existing commands that have the same name as commands being installed by a module. AllowClobber and Force can be used together in an Install-Module command." [https://docs.microsoft.com/en-us/powershell/module/powershellget/install-module?view=powershell-6]

Reply:
> Yes, but you also need to consider to be able to work with PowerShell 5.1 as it is mostly used on the machines still

I've got PSv5.1 and it has the `-AllowClobber` parameter, AFAIK it's based on the Package Management/PowerShell Get version installed so maybe the one that comes with PSv5.1 out of the box isn't new enough.

Still in saying that the code right now shouldn't be working because we set `MinimumVersion` in `$install_params` which is not a valid parameter for `Update-Module`.

Reply:
You can use the Update-Module with the following syntax which means the Version: 
(e.g. Update-Module -Name SpeculationControl -RequiredVersion 1.0.14)

Reply:
See also there: https://docs.microsoft.com/en-us/powershell/module/powershellget/update-module?view=powershell-5.1

Reply:
What I'm trying to say is [Update-Module](https://docs.microsoft.com/en-us/powershell/module/powershellget/update-module?view=powershell-5.1) does **NOT** have a parameter `-MinimumVersion` but we are calling it with the `@install_params` splat which **DOES** set that parameter https://github.com/ansible/ansible/pull/63820/files#diff-f5690a225345237d5b24a08ebf47102cR74.

This means that if we ever execute the `Update-Module` path is probably is going to fail because we are calling it with a parameter that it does not support.